src/modals/storage-connections/StorageConnectionsDataProvider.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import getPluginApi from '_/plugin-api'
+import { engineGet, enginePost, enginePut, engineDelete } from '_/utils/fetch'
+import DataProvider from '_/components/helper/DataProvider'
+import { webadminToastTypes } from '_/constants'
+import { msg } from '_/intl-messages'
+
+const createConnection = async (connection) => {
+  const response = await enginePost(
+    'api/storageconnections',
+    JSON.stringify(connection)
+  )
+  return response
+}
+
+const editConnection = async (connection, connectionId) => {
+  const response = await enginePut(
+    `api/storageconnections/${connectionId}`,
+    JSON.stringify(connection)
+  )
+  return response
+}
+
+const deleteConnection = async (connectionId) => {
+  const response = await engineDelete(`api/storageconnections/${connectionId}`)
+  return response
+}
+
+const attachConnection = async (connectionId, domainId) => {
+  const response = await enginePost(
+    `api/storagedomains/${domainId}/storageconnections`,
+    JSON.stringify({ id: connectionId })
+  )
+  return response
+}
+
+const detachConnection = async (connectionId, domainId) => {
+  const response = await engineDelete(`api/storagedomains/${domainId}/storageconnections/${connectionId}`)
+  return response
+}
+
+const fetchData = async (storageDomainId) => {
+  const allConnectionsJson = await engineGet('api/storageconnections')
+  let storageDomain = await engineGet(`api/storagedomains/${storageDomainId}`)
+  const domainConnectionsJson = await engineGet(`api/storagedomains/${storageDomainId}/storageconnections`)
+
+  const dataCenter = storageDomain.data_centers.data_center[0]
+  if (!dataCenter) {
+    throw new Error('StorageConnectionsDataProvider: failed to fetch data center')
+  }
+
+  storageDomain = await engineGet(`api/datacenters/${dataCenter.id}/storagedomains/${storageDomainId}`)
+
+  if (!storageDomain) {
+    throw new Error('StorageConnectionsDataProvider: failed to fetch storage domain')
+  }
+
+  let allConnections = allConnectionsJson.storage_connection
+  let domainConnections = domainConnectionsJson.storage_connection
+
+  if (!allConnections || allConnections.error) {
+    throw new Error('StorageConnectionsDataProvider: failed to fetch storage connections')
+  }
+
+  if (!domainConnections || domainConnections.error) {
+    throw new Error('StorageConnectionsDataProvider: failed to fetch storage connections' +
+      'for storage domain ' + storageDomainId)
+  }
+
+  const domainConnectionsIds = new Set(domainConnections.map(connection => connection.id))
+  const isConnectionAttachedToDomain = (conn) => { return domainConnectionsIds.has(conn.id) }
```

Comment:
It seems a bit like overhead to create a separate function for such a simple condition.

Reply:
Yes, but the lines that these conditions are used are already seemed to me quite cumbersome, so I thought it would improve the readability of the code